ksqldb-engine/src/test/java/io/confluent/ksql/query/KafkaStreamsQueryValidatorTest.java

```diff
@@ -138,6 +149,38 @@ public void shouldIgnoreGlobalLimitSetInOverrides() {
     );
   }
 
+  @Test
```

Comment:
Do we have a test for the mix of BinPackedPersistentQueryMetadataImpl and non-binpacked metadata impl to test the summing behavior above, that when a new binpacked query is added we may violate the limit?

Reply:
This is already testing that actually. We have 2 old queries and 2 new qureies that are in the same runtime